pkg/ocm/upgrades.go

```diff
@@ -96,3 +98,79 @@ func (c *Client) CancelUpgrade(clusterID string) (bool, error) {
 	}
 	return true, nil
 }
+
+func (c *Client) GetMissingGateAgreements(
+	clusterID string,
+	upgradePolicy *cmv1.UpgradePolicy) ([]*cmv1.VersionGate, error) {
+	response, err := c.ocm.ClustersMgmt().V1().
+		Clusters().Cluster(clusterID).
+		UpgradePolicies().
+		Add().
+		Parameter("dryRun", true).
+		Body(upgradePolicy).
+		Send()
+	if err == nil {
+		return []*cmv1.VersionGate{}, nil
+	}
+
+	// parse gates list
+	details, ok := response.Error().GetDetails()
```

Comment:
@pvasant @vkareh @jhernand @tzvatot I really don't like the following solution.

For our purpose, it would be much better to add a query endpoint to CS with a proprietary API in the SDK like `GetMissingGateAgreements()`, and that would return the list of gates.

However, since we decided to rely on `Error().Details()` I can't use the returned map to read the object, since the sdk doesn't have json annotaion for the fields, so simplest way was to just read the fields we need from the map.
That was the quickest solution.

If we decide to add the API to CS and the SDK, we can change this

Reply:
You can marshal the details to JSON and then unmarshal the type you need:

```go
data, err := json.Marshal(details)
if err != nil {
        ...
}
gates := cmv1.UnmarshalVersionGateList(details)
if err != nil {
        ...
}
```